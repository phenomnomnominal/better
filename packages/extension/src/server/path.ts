import type { TextDocument } from 'vscode-languageserver-textdocument';

import { URI } from 'vscode-uri';

import { isString } from '../utils';

export function getFilePath(documentOrUri: URI | TextDocument | string): string | null {
  if (!documentOrUri) {
    return null;
  }
  let uri = null;
  if (documentOrUri instanceof URI) {
    uri = documentOrUri;
  } else if (isString(documentOrUri)) {
    uri = URI.parse(documentOrUri);
  } else {
    uri = URI.parse(documentOrUri.uri);
  }
  if (uri.scheme !== 'file') {
    return null;
  }
  return uri.fsPath;
}
