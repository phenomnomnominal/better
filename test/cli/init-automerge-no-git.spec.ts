import { describe, it, expect } from 'vitest';
import path from 'node:path';

import { createFixture } from '../fixture.js';

const ARGV = ['node', './bin/betterer'];

describe('betterer cli', () => {
  it('should enable automerge configuration without git', async () => {
    const { cliÎ” } = await import('@betterer/cli');

    const { cleanup, logs, paths } = await createFixture(
      'init-automerge-no-git',
      {
        './package.json': `
      {
        "name": "init-automerge",
        "version": "0.0.1"
      }
      `
      },
      {
        logFilters: [/ðŸŒŸ Initialising Betterer/, /: running /, /running.../]
      }
    );

    const fixturePath = paths.cwd;

    process.env.BETTERER_WORKER = 'false';

    await cliÎ”(fixturePath, [...ARGV, 'init', '--automerge', `--repoPath=${path.sep}`]);

    expect(logs).toMatchSnapshot();

    await cleanup();
  });
});
