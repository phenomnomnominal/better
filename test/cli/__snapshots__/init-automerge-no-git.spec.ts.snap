// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`betterer cli > should enable automerge configuration without git 1`] = `
[
  "ðŸ’¥ Initialising Betterer (0ms): 3 tasks done! 1 task errored!
âœ… Create test file: done!
ãƒ» created "fixtures/init-automerge-no-git/.betterer.ts"!
âœ… Update package.json: done!
ðŸ”¥ Enable automerge: ".git" directory not found in "/". \`--automerge\` only works within a Git repository.

Error: ".git" directory not found in "/". \`--automerge\` only works within a Git repository.",
  "ðŸ’¥ Initialising Betterer (0ms): 3 tasks done! 1 task errored!
âœ… Create test file: done!
ãƒ» created "fixtures/init-automerge-no-git/.betterer.ts"!
âœ… Update package.json: done!
ãƒ» added "betterer" script to package.json file
ãƒ» added "@betterer/cli" dependency to package.json file
ãƒ» added "typescript" dependency to package.json file
ðŸ”¥ Enable automerge: ".git" directory not found in "/". \`--automerge\` only works within a Git repository.

Error: ".git" directory not found in "/". \`--automerge\` only works within a Git repository.",
]
`;
